### DISCUSSION TITLE
Universal scripts handling

### DISCUSSION THREAD

**Discussion by rvisharma** [johnlindquist/kit#263] [Q&A] [answered]
vague title maybe? 😛

Hellooo,
How do you manage scripts that need to be executed both via app and terminal

For eg,
```js
await arg('just press enter')

// this doesn't fail in terminal
setPlaceholder('placeholder')

// all of these below fails in terminal
// UNCAUGHT EXCEPTION: ReferenceError: setInput is not defined
setInput('input input')
setPanel('panel panel')
setHint('hint hint')

await wait(2_000)

```

This script works well on App, but fails on terminal because few api are unavailable on terminal (which makes sense) like setHint, setInput, setPanel.
Also, setPlaceholder doesn't throw exception somehow.

My current approach is to use something like `setXXX && setXXX()`

- Are there any other ways to handle it gracefully?
- Should these unavailable api just be a no-op on cli environment?

**Comment by johnlindquist** [comment, marked as answer]
`env.KIT_CONTEXT` is set to "app" when a script is run from the app.

```js
if(env.KIT_CONTEXT === "app"){
//app specific stuff
}
```

I'm still trying to think how `setHint`, etc can be brought into the terminal environment, but I haven't had much time to focus on the terminal experience recently. It's definitely something I'm looking forward to.

**Reply by rvisharma** [reply]
Thanks! This would do.
